component/main.jsonnet

```diff
@@ -1,10 +1,110 @@
 // main template for csi-driver-smb
+local com = import 'lib/commodore.libjsonnet';
+local espejo = import 'lib/espejo.libsonnet';
 local kap = import 'lib/kapitan.libjsonnet';
 local kube = import 'lib/kube.libjsonnet';
 local inv = kap.inventory();
 // The hiera parameters for the component
 local params = inv.parameters.csi_driver_smb;
 
-// Define outputs below
+local pvs = [
+  kube.PersistentVolume(pv.name) {
+    metadata+: {
+      namespace: pv.namespace,
+    },
+    spec+: {
+      capacity: {
+        storage: pv.capacity,
+      },
+      accessModes: com.getValueOrDefault(pv, 'accessModes', [ 'ReadWriteMany' ],),
+      persistenVolumeReclaimPolicy: com.getValueOrDefault(pv, 'reclaimPolicy', null),
+      mountOptions: com.getValueOrDefault(pv, 'mountOptions', [ 'dir_mode=0777', 'file_mode=0777', 'vers=3.0' ]),
+      csi: {
+        driver: 'smb.csi.k8s.io',
+        readOnly: com.getValueOrDefault(pv, 'readOnly', null),
+        volumeAttributes: {
+          source: '//%s/%s' % [ pv.share_host, pv.share_name ],
+        },
+        volumeHandle: 'pv-%s-%s' % [ pv.namespace, pv.name ],
+        nodeStageSecretRef: {
+          name: '%s-credentials' % [ pv.name ],
+          namespace: pv.namespace,
+        },
+      },
+      storageClassName: 'smb-%s' % [ pv.namespace ],
```

Comment:
Where is this storage class coming from? Besides it looks like there can only be 1 storage class resp. share per namespace. Doesn't it make more sense to provide the class name from e.g. `pv.storageClass` parameter?

The [documentation](https://github.com/kubernetes-csi/csi-driver-smb/blob/master/deploy/example/e2e_usage.md) mentions using either a storageclass OR a specifying the share parameters in the PV (but without storageclass param). I think we should configure the PV either with built-in CSI spec, or using the storage class, right? (I don't know how it works with SMB driver but that's my guess)

Using the option to use storage classes, one would not need to create a PV separately afaik, just define a PVC with storage class like https://raw.githubusercontent.com/kubernetes-csi/csi-driver-smb/master/deploy/example/statefulset.yaml

Reply:
You can add `storageClassName` to a PV without the StorageClass itself existing. This is useful, e.g. to limit who can claim the PV with a `ResourceQuota` based on the storage class name, cf. https://kubernetes.io/docs/concepts/policy/resource-quotas/#storage-resource-quota